packages/edit-site/src/store/actions.js

```diff
@@ -211,11 +211,17 @@ export function* setPage( page ) {
 		// If the entity is undefined for some reason, path will resolve to "/"
 		page.path = getPathAndQueryString( entity?.link );
 	}
-	const { id: templateId, slug: templateSlug } = yield controls.resolveSelect(
+	const template = yield controls.resolveSelect(
 		coreStore,
 		'__experimentalGetTemplateForLink',
 		page.path
 	);
+
+	if ( ! template ) {
+		return;
```

Comment:
Should there be any kind of visible error for users if there's no template?

If so, perhaps a notice can be dispatched.

Reply:
The editor will display a notice if the template doesn't exist:

https://github.com/WordPress/gutenberg/blob/9543b72e701a452858b1434977275bc3fbf3a8bd/packages/edit-site/src/components/editor/index.js#L253-L267

Reply:
Great, in that case I've approved the PR.